assets/opstools/BuildApp/controllers/data_fields/ageequation.js

```diff
@@ -0,0 +1,388 @@
+steal(function () {
+	var componentIds = {
+		editView: 'ab-new-age',
+		name: 'ab-new-age-name',
+		equaltionType: 'ab-new-age-equaltionType',
+		dateType: 'ab-new-age-dateType',
+		resultType: 'ab-new-age-resultType',
+		equation: 'ab-new-age-equation',
+		typeDecimals: 'ab-new-age-typeDecimals',
+		typeDecimalPlaces: 'ab-new-age-typeDecimalPlaces',
+		typeRounding: 'ab-new-age-typeRounding',
+		typeThousands: 'ab-new-age-typeThousands',
+		typeFormat: 'ab-new-age-typeFormat',
+
+
+
+	};
+	
+	var formatList = [
+		{ id: 'none', value:  "None" },
+		{ id: 'dollar', value:  "$", sign: "$", position: "prefix" },
+		{ id: 'pound', value: "£", sign: "£", position: "prefix" },
+		{ id: 'euroBefore', value: "€ (before)", sign: "€", position: "prefix" },
+		{ id: 'euroAfter', value:  "€ (after)", sign: "€", position: "postfix" },
+		{ id: 'percent', value:  "%", sign: "%", position: "postfix" },
+	];
+	
+	var ageequationDataField = {
+		name: 'ageequation', // unique key to reference this specific DataField
+		type: 'string', // http://sailsjs.org/documentation/concepts/models-and-orm/attributes#?attribute-options
+		icon: 'font',   // font-awesome icon reference.  (without the 'fa-').  so 'user'  to reference 'fa-user'
+	
+		menuName: 'Age equation',  
+		includeHeader: true,
+		description: ''
+	};
+	
+	// datepart: 'y', 'm', 'w', 'd', 'h', 'n', 's'
+	function caldateDiff(datepart, fromdate, todate){
+		
+		  datepart = datepart.toLowerCase();	
+			  var diff = todate - fromdate;	
+			  var divideBy = { weeks:604800000, 
+				days:86400000, 
+				hours:3600000, 
+				years:31557600000};	
+
+			return Math.floor( diff/divideBy[datepart]);
+
+	}
+	
+	function showsettingNumeric(){
+		$$(componentIds.dateType).hide();
+		$$(componentIds.resultType).hide();
+	}
+	
+	function showsettingDate(){
+		$$(componentIds.dateType).show();
+		$$(componentIds.resultType).show();
+	}
+	
+	function equationValidate(text){
+		//{currentdate} - {birthdate} 
+	}
+	
+
+
+
+	ageequationDataField.editDefinition = {
+		id: componentIds.editView,
+		rows: [
+			{
+				view: "text",
+				label: "Name",
+				labelWidth: "100",
+				id: componentIds.name,
+				placeholder: 'Age'
+			},
+			{
+				view: "radio",
+				id: componentIds.equaltionType,
+				label: "Equaltion Type",
+				labelWidth: "110",
+				value: 'none',
+				vertical: true,
+				options: [
+					{ id: 'numeric',value: "Numeric" },
+					{ id: 'date',value:  "Date" },
+					
+				],
+				on: {
+					'onChange': function (newValue, oldValue) {
+						if(newValue == 'numeric'){
+							showsettingNumeric();
+						}else{
+							showsettingDate();
+						}
+						
+						
+					}
+				}
+			},
+			{
+				cols: [
```

Comment:
It looks like we are redefining the settings for the date and number data fields here.

Let's see if we can simply reuse those settings from the actual date.js and number.js here.

This would also imply that when it is time to display the result of this equation, we are reusing the date.js or number.js datafield to display our results.